src/Microsoft.Extensions.Logging.Testing/AssemblyTestLog.cs

```diff
@@ -84,7 +86,45 @@ public IServiceProvider CreateLoggerServices(ITestOutputHelper output, string cl
             SerilogLoggerProvider serilogLoggerProvider = null;
             if (!string.IsNullOrEmpty(_baseDirectory))
             {
-                var testOutputFile = Path.Combine(_baseDirectory, _assemblyName, RuntimeInformation.FrameworkDescription.TrimStart('.'), className, $"{testName}.log");
+                var testOutputDirectory = Path.Combine(_baseDirectory, _assemblyName, RuntimeInformation.FrameworkDescription.TrimStart('.'), className);
+
+                if (testOutputDirectory.Length + testName.Length + LogFileExtension.Length >= MaxPathLength)
+                {
+                    output.WriteLine($"Warning: test name {testName} is too long. Please shorten test name.");
+
+                    // Shorten the test name by removing the middle portion of the testname
+                    var testNameLength = MaxPathLength - testOutputDirectory.Length - LogFileExtension.Length;
+
+                    if (testNameLength <= 0)
+                    {
+                        throw new InvalidOperationException("Output file path could not be constructed due to max path length restrictions. Please shorten test assembly, class or method names.");
+                    }
+
+                    testName = testName.Substring(0, testNameLength / 2) + testName.Substring(testName.Length - testNameLength / 2, testNameLength / 2);
+                }
+
+                var testOutputFile = Path.Combine(testOutputDirectory, $"{testName}{LogFileExtension}");
+
+                if (File.Exists(testOutputFile))
+                {
+                    output.WriteLine($"Warning: Output log file {testOutputFile} already exists. Please try to keep log file names unique.");
+
+                    for (var i = 0; i < 1000; i++)
```

Comment:
This should be rare, and exceptional (i.e. the test isn't properly named) right?

Reply:
I think it will happen quite often as theory based tests might not end up with unique names. For example if one of the parameter is a string and the test uses string.Empty and null as parameters. It is very easy to overlook but we still probably want to capture the logs. The big downside to this is that if you run the same test repeatedly, the old log files will no longer be overwritten and you'll end up with a long list of test files.